shell_automaton/src/protocol_runner/protocol_runner_effects.rs

```diff
@@ -103,6 +107,37 @@ where
                             continue;
                         }
                     },
+                    ProtocolRunnerState::GetCurrentHead(state) => match state {
+                        ProtocolRunnerCurrentHeadState::Pending { .. } => match result {
+                            ProtocolRunnerResult::GetCurrentHead((
+                                token,
+                                Ok(latest_context_hashes),
+                            )) => {
+                                store.dispatch(ProtocolRunnerCurrentHeadSuccessAction {
+                                    token,
+                                    latest_context_hashes,
+                                });
+                                continue;
+                            }
+                            ProtocolRunnerResult::GetCurrentHead((token, Err(error))) => {
+                                slog::error!(&store.state().log, "failed to get context's latest commits";
+                                    "error" => format!("{:?}", error));
+                                store.dispatch(ProtocolRunnerCurrentHeadSuccessAction {
```

Comment:
Shouldn't you dispatch error here? Probably you are dispatching success action because even if getting latest context hashes fails, you want it to work. But you still can do that by dispatching error and afterwards doing exact same thing as in case of success.

Reply:
Thanks, I will do that